src/Microsoft.Azure.SignalR/ServiceOptionsSetup.cs

```diff
@@ -35,6 +38,21 @@ public ServiceOptionsSetup(IConfiguration configuration)
 
             _connectionString = connectionString;
             _endpoints = endpoints;
+
+            _gracefulShutdownEnabled = configuration[Constants.ServerGracefulShutdownKey] == "Enabled" ? true : false;
```

Comment:
How about separating the PR into several, for example, one for adding and reading options, one for adding interface, one for the drain....

Reply:
I'd like to remain the compactness of this PR since this feature doesn't have any potential risk I guess?